### PR TITLE
feat(rdb): add read_replica resource

### LINKED ISSUE #1409: Add support for read replicas
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### New or Affected Resource(s)

* scaleway_XXXXX

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

* #0000

### PR DESCRIPTION
**Pull request by Codelax**
Fix #1409

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -250,6 +250,8 @@ github.com/robfig/cron/v3 v3.0.1/go.mod h1:eQICP3HwyT7UooqI/z+Ov+PtYAWygg1TEWWzG
 github.com/rogpeppe/fastuuid v1.2.0/go.mod h1:jVj6XXZzXRy/MSR5jhDC/2q6DgLz+nrA6LYCDYWNEvQ=
 github.com/scaleway/scaleway-sdk-go v1.0.0-beta.9.0.20220801131554-e792c0d04360 h1:AJqOOYKhXG4GGetUAKP/44CmqCkaP0aYdL/IvR1Zeq4=
 github.com/scaleway/scaleway-sdk-go v1.0.0-beta.9.0.20220801131554-e792c0d04360/go.mod h1:fCa7OJZ/9DRTnOKmxvT6pn+LPWUptQAmHF/SBJUGEcg=
+github.com/scaleway/scaleway-sdk-go v1.0.0-beta.9.0.20220805142944-61ee3b46cb4b h1:IG4ZfjvInA8rrMqVWRFZzQsFybU6KOHIoIsS/WFnaj0=
```
Comment (remyleone) [inline go.sum:253]: go mod tidy

File: scaleway/provider.go [hunk 2]
```diff
@@ -137,6 +137,7 @@ func Provider(config *ProviderConfig) plugin.ProviderFunc {
 				"scaleway_rdb_instance":                        resourceScalewayRdbInstance(),
 				"scaleway_rdb_privilege":                       resourceScalewayRdbPrivilege(),
 				"scaleway_rdb_user":                            resourceScalewayRdbUser(),
+				"scaleway_rdb_read_replica":                    resourceScalewayRdbReadReplica(),
```
Comment (Monitob) [inline scaleway/provider.go:140]: Please add the documentation about this

File: scaleway/resource_rdb_read_replica.go [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+			"endpoints": {
+				Type:     schema.TypeList,
+				MaxItems: 1,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"direct_access": {
+							Type:        schema.TypeList,
+							Optional:    true,
+							Description: "Direct access endpoint",
```
Comment (Monitob) [inline scaleway/resource_rdb_read_replica.go:46]: What's direct access mean?

File: scaleway/resource_rdb_read_replica.go [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									// Private network specific
+									"private_network_id": {
+										Type:         schema.TypeString,
+										Description:  "UUID of the private network to be connected to the read replica (UUID format).",
+										ValidateFunc: validationUUIDorUUIDWithLocality(),
+										Required:     true,
+									},
+									"service_ip": {
```
Comment (Monitob) [inline scaleway/resource_rdb_read_replica.go:93]: Add a validator here

File: scaleway/resource_rdb_read_replica.go [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+									},
+									"hostname": {
+										Type:        schema.TypeString,
+										Description: "Hostname of the endpoint. Only one of ip and hostname may be set.",
+										Computed:    true,
+									},
+								},
+							},
+						},
+						"private_network": {
```
Comment (Monitob) [inline scaleway/resource_rdb_read_replica.go:79]: Can we have `direct_access` and `private_network`? Maybe add Conflict with

**Review by Monitob — CHANGES_REQUESTED** [review]

File: scaleway/resource_rdb_read_replica.go (hunk 5 above)
Comment (Codelax) [reply scaleway/resource_rdb_read_replica.go:80]: We can have both

**Review by Monitob — APPROVED** [review]

**Pull request merged**